lib/esLoader.js

```diff
@@ -60,62 +62,98 @@ function load(options, client, basename, callback){
   }
 
 
-  createIndex(options, client, basename, function(err, index){
+  createIndex(options, basename, function(err, index){
     if(err) return callback(err);
 
     var stream = new writeStream(client, index, alias, type);
 
     stream.on('finish', function(){
-      var self = this;
+      applyAlias(options, index, function(err){
+        if(err) return stream.emit('error', err);
+        return stream.emit('loaded');
+      })
+    });
 
-      client.indices.get({index: alias}, function(err, res){
-        if(err){
-          if(err.message.slice(0, 21) === 'IndexMissingException'){
-            res = [];
-          }else{
-            return stream.emit('error', err);
-          }
-        }
+    return callback(null, stream);
+  });
+}
 
-        var internalIndexArr = Object.keys(res);
-        var oldIndices = [];
 
-        for(var i=0; i<internalIndexArr.length; i++){
-          var index = internalIndexArr[i];
-          if(index.split('-').slice(0, -2).join('-') === basename){
-            oldIndices.push(index);
-          }
-        }
-
-        client.indices.putAlias({
-          index: self.index,
-          name: self.alias
-        }, function(err){
-          if(err) return stream.emit('error', err);
-
-          if(oldIndices.length){
-            client.indices.delete({index: oldIndices.join(',')}, function(err){
-              if(err && err.message.slice(0, 21) !== 'IndexMissingException'){
-                stream.emit('error', err)
-              }else{
-                client.close();
-                stream.emit('alias');
-              }
-            });
+function loadIntoIndex(options, callback){
+
+  var client = options.client;
+  var alias = options.alias;
+  var index = options.forcedIndex;
```

Comment:
For tiger I am now forcing every file into the same internal index, rather than creating separate internal indices for each file.

There was quite a bit of overhead associated with creating all the extra internal indices as well as ulimit errors which could cause interruptions